zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -521,11 +525,16 @@ private void receiveMessage(ByteBuf message) {
                             throw new IOException("Len error " + len);
                         }
                         ZooKeeperServer zks = this.zkServer;
-                        if (zks == null || !zks.isRunning()) {
-                            throw new IOException("ZK down");
+
+                        if (zks != null && !zks.isRunning()) {
+                            // checkRequestSize will throw IOException if request is rejected
+                            zks.checkRequestSizeWhenReceivingMessage(len);
+                        } else {
+                            LOG.debug("Skip configurable check for max receive length");
+                            if (len > 64 * 1024) {
+                                throw new IOException("Received message size too large for uninitialized server");
+                            }
```

Comment:
This check looks unnecessary. 
There's already https://github.com/apache/zookeeper/blob/c926b2f4d58ad08c33ff88ff6256347dab77ef60/zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxn.java#L524-L526